src/Utilities/LockCheck.cs

```diff
@@ -247,19 +251,31 @@ internal static string GetProcessesLockingFile(string filePath)
         }
 
         /// <summary>
-        /// Try to get a message to inform the user which processes have a lock on a given file.
+        /// Try to get a message to inform the user which processes have a lock on a given file. On Windows it uses the Restart Manager API.
         /// </summary>
-        internal static string GetLockedFileMessage(string file)
+        /// <param name="filePath">The path of the file to check.</param>
+        /// <returns>A message to inform the user which processes have a lock on the file on Window if available, string.Empty on other platforms.</returns>
+        public static string GetLockedFileMessage(string filePath)
+        {
+            if (NativeMethodsShared.IsWindows)
+            {
+                return GetLockedFileMessageWindows(filePath);
+            }
+            return string.Empty;
+        }
+
+        [SupportedOSPlatform("windows")]
+        private static string GetLockedFileMessageWindows(string filePath)
```

Comment:
> can't this logic be a part of GetLockedFileMessage?
I split it so that there can be the [SupportedOSPlatform("windows")] annotation for `GetLockedFileMessageWindows` and the public `GetLockedFileMessage` which checks for windows first is extensible in case we wanted to support other platforms. 
other motivation for splitting was moving the `if` from the callers to `LockCheck` class but don't make it super-nested.